homeassistant/components/govee_light_api/config_flow.py

```diff
@@ -0,0 +1,126 @@
+"""Config flow for Govee Local API."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components import network
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_BIND_ADDRESS,
+    CONF_DISCOVERY_INTERVAL,
+    CONF_DISCOVERY_INTERVAL_DEFAULT,
+    CONF_LISENING_PORT,
+    CONF_LISENING_PORT_DEFAULT,
+    CONF_MULTICAST_ADDRESS,
+    CONF_MULTICAST_ADDRESS_DEFAULT,
+    CONF_TARGET_PORT,
+    CONF_TARGET_PORT_DEFAULT,
+    DOMAIN,
+)
+
+CONFIG_SCHEMA_ADVANCED = {
+    vol.Required(
+        CONF_MULTICAST_ADDRESS,
+        default=CONF_MULTICAST_ADDRESS_DEFAULT,
+    ): cv.string,
+    vol.Required(CONF_LISENING_PORT, default=CONF_LISENING_PORT_DEFAULT): cv.port,
+    vol.Required(CONF_TARGET_PORT, default=CONF_TARGET_PORT_DEFAULT): cv.port,
+}
+
+
+_LOGGING = logging.getLogger(__name__)
+
+
+class GoveeConfigFlow(ConfigFlow, domain=DOMAIN):
```

Comment:
Since we're only doing UDP discovery, isn't a config flow like Gree better? Are there reasons why someone has a different port for UDP discovery?

Reply:
Yeah, probably chainging the ports (or even the multicast address) is not necessary, since they cannot be changed on the Govee side.
I'll check the Gree flow and update this one

Reply:
Regarding this, I'm almost done, but I didn't find a way to keep the OptionsFlow.
Is there a way to keep it? It's not really needed but we will lose the possibility to change the discovery interval.

Also, looking at the Gree integration, I had a couple of ideas:
* Would be better to fire an event when a device is discovered?
* Currently the discovery it is done using the `govee_local_api` library (internally it uses a `call_later` on the same Home Assistant event loop) but, since this can be disabled, could be better if the update function is triggered using something like `async_track_time_interval`?

Reply:
Why would one change the discovery interval?

What do you mean with the config flow can be disabled?

Reply:
Also, Optionsflow should be in a follow up PR. If with the gree config flow no options flow is possible and we still want one, I'd rather fix something in core so we are able to add an options flow

Reply:
> Why would one change the discovery interval?

That is probably not really needed, unless maybe for debug purposes.

> What do you mean with the config flow can be disabled?

I didn't mean the config flow, I mixed up things there.
What I meant is that the library internally uses `call_later` to poll the Govee devices and update their status, but this can be disabled and the devices can be polled "manually".
I was wondering if it should be better to let Home Assistant call the poll function.